install/first_google_boot.sh

```diff
@@ -224,11 +224,22 @@ function extract_spinnaker_kube_credentials() {
   chown -R spinnaker:spinnaker $(dirname $config_path)
 
   local kube_cluster=$(get_instance_metadata_attribute "kube_cluster")
+  local kube_project=$(get_instance_metadata_attribute "kube_project")
+  local kube_zone=$(get_instance_metadata_attribute "kube_zone")
+
+  if [ -z "$kube_project" ]; then
+    kube_project=$MY_PROJECT
+  fi
+
+  if [ -z "$kube_zone" ]; then
+    kube_zone=$MY_ZONE
+  fi
+
   if [ -n "$kube_cluster" ]; then
```

Comment:
spinnaker.yml is just a file. Anyone can read it at any time. The thing producing the spinnaker.yml can also be the thing suppling this metadata. It's just a packaging.

The immutable infrastructure argument is a good point that probably negates this. At least for production use cases. For development, I anticipate it to be annoying. Like creating a VM then realizing you left a scope out and having to trash it and create another one (and wait for  the creation, spinup, etc).